src/rdkafka_partition.c

```diff
@@ -3013,6 +3072,22 @@ int rd_kafka_topic_partition_cmp(const void *_a, const void *_b) {
                 return RD_CMP(a->partition, b->partition);
 }
 
+/**
+ * @brief Compare topic partitions \p a and \p b by topic id first
+ *        and then by partition.
+ */
+int rd_kafka_topic_partition_by_id_cmp(const void *_a, const void *_b) {
+        const rd_kafka_topic_partition_t *a = _a;
+        const rd_kafka_topic_partition_t *b = _b;
+        rd_kafka_Uuid_t topic_id_a = rd_kafka_topic_partition_get_topic_id(a);
+        rd_kafka_Uuid_t topic_id_b = rd_kafka_topic_partition_get_topic_id(b);
+        int r                      = rd_kafka_Uuid_cmp(topic_id_a, topic_id_b);
```

Comment:
Better name for `r` so that it is more readable in next statements. Maybe `are_topic_ids_equal`?